### ISSUE TITLE
AlfDynamicPayloadButton should allow subscription to topics from other scopes

### ISSUE THREAD

**Issue by AFaust** [Alfresco/Aikau#1096]
In my project to Aikau-ify the JavaScript Console addon I encountered a limitation of the AlfDynamicPayloadButton that I can't work around without crazy scope-hacking. Basically, the button needs to mix in data from multiple different topics (collecting JS/FTL source from CodeMirror instances, execution parameters from a dynamic form), and at least one of those topics will always be published in a different pubSubScope than the scope of the button. Unfortunately, it is not possible to define an alternative subscription scope or just "subscribe to global" for the conigured ```publishPayloadSubscriptions```. This currently means I cannot mixin form values since the form has (and must have) a custom pubSubScope to isolate it from potentially other forms that may be added to the console tool.

Generally all configurations that lead to a component / feature relying on subscription should allow subscription to global, parent or a custom scope. In this instance I suggest the AlfDynamicPayloadButton to support coniguration along the lines of:
```javascript
{
   name : 'alfresco/buttons/AlfDynamicPayloadButton',
   config : {
      label : 'Click me',
      publishTopic : MY_PUBLISH_TOPIC,
      publishGlobal : true,
      publishPayload : {},
      publishPayloadSubscriptions : [ {
         // topic always required
         topic : 'MY_SUBSCRIPTION_TOPIC',
         // subscribeGlobal true/false = ultimate preference over subscribeParent, custom scope and default widget scope
         subscribeGlobal : true,
         // subscribeParent true/false = supercedes custom scope and default widget scope
         subscribeParent : true,
         // subscribeScope = alternative to default widget scope
         subscribeScope : 'MY_SCOPE',
         dataMapping : {
            myProperty : 'path.to.payloadProperty'
         }
      }]
   }
}
```

Similarily the CoreWidgetProcessing (in)visibilityConfig rules should allow optionally specifying the subscription scope. The ```alfConditionalSubscribe``` operation already supports global and parentScope, and would only need to be enhanced for passing a custom scope to ```alfSubscribe```. (This is one of the other local enhancements we made to CoreWidgetProcessing alluded to in #1047).

This is my current button configuration for reference: https://github.com/AFaust/js-console/blob/js-console-v2/javascript-console-share-aikau/src/main/amp/web/js/javascript-console/tool/JavaScriptConsoleTool.js#L96 (introduced in commit https://github.com/AFaust/js-console/commit/cbe2d3a577d2fdd2024b9e48604cb59aae9c2ed1).

**Comment by draperd** [comment]
I've raised https://issues.alfresco.com/jira/browse/AKU-987 to address this. We'll put it into the next sprint so the updates should hopefully be available in the next release (1.0.71) - however, the team has reduced capacity this week so there isn't any guarantee that it will be completed this sprint but we'll do our best.

**Comment by draperd** [comment]
This is now available in the 1.0.71 release
